executor/aggfuncs/window_func_test.go

```diff
@@ -151,3 +164,48 @@ func (s *testSuite) TestWindowFunctions(c *C) {
 		s.testWindowFunc(c, test)
 	}
 }
+
+func (s *testSuite) TestMemRank(c *C) {
+	tests := []aggMemTest{
+		buildAggMemTester(ast.WindowFuncRank, mysql.TypeLonglong, 5,
+			aggfuncs.DefPartialResult4RankSize, rowMemDeltaGens, false),
+		buildAggMemTester(ast.WindowFuncRank, mysql.TypeFloat, 5,
+			aggfuncs.DefPartialResult4RankSize, rowMemDeltaGens, false),
+		buildAggMemTester(ast.WindowFuncRank, mysql.TypeDouble, 5,
+			aggfuncs.DefPartialResult4RankSize, rowMemDeltaGens, false),
+		buildAggMemTester(ast.WindowFuncRank, mysql.TypeNewDecimal, 5,
+			aggfuncs.DefPartialResult4RankSize, rowMemDeltaGens, false),
+	}
+	for _, test := range tests {
+		s.testWindowAggMemFunc(c, test)
+	}
+}
+
+func (s *testSuite) testWindowAggMemFunc(c *C, p aggMemTest) {
+	srcChk := p.aggTest.genSrcChk()
+
+	args := []expression.Expression{&expression.Column{RetType: p.aggTest.dataType, Index: 0}}
+	if p.aggTest.funcName == ast.AggFuncGroupConcat {
+		args = append(args, &expression.Constant{Value: types.NewStringDatum(" "), RetType: types.NewFieldType(mysql.TypeString)})
+	}
+	desc, err := aggregation.NewAggFuncDesc(s.ctx, p.aggTest.funcName, args, p.isDistinct)
+	c.Assert(err, IsNil)
+	if p.aggTest.orderBy {
+		desc.OrderByItems = []*util.ByItems{
+			{Expr: args[0], Desc: true},
+		}
+	}
+	finalFunc := aggfuncs.BuildWindowFunctions(s.ctx, desc, 0, nil)
+	finalPr, memDelta := finalFunc.AllocPartialResult()
+	c.Assert(memDelta, Equals, p.allocMemDelta)
+
+	updateMemDeltas, err := p.updateMemDeltaGens(srcChk, p.aggTest.dataType)
+	c.Assert(err, IsNil)
+	iter := chunk.NewIterator4Chunk(srcChk)
+	i := 0
+	for row := iter.Begin(); row != iter.End(); row = iter.Next() {
+		memDelta, _ := finalFunc.UpdatePartialResult(s.ctx, []chunk.Row{row}, finalPr)
```

Comment:
do we need to check for errors here?

Reply:
I found the same problem in the aggfunc_test and fixed it in the new commit.